consent/manager_test_helpers.go

```diff
@@ -571,120 +590,185 @@ func ManagerTests(m Manager, clientManager client.Manager, fositeManager x.Fosit
 			})
 
 			t.Run("case=ListUserAuthenticatedClientsWithFrontChannelLogout", func(t *testing.T) {
-				for i := 0; i <= 10; i++ {
-					c, h := MockConsentRequest(fmt.Sprintf("LUACWFCL-%d", i), false, 0, false, false, false)
-					if i == 5 {
-						c.Client.FrontChannelLogoutURI = "http://some-url.com/"
-					}
-					c.LoginSessionID = "" // otherwise we had to create the login session as well..
-					c.Subject = "subjectLUACWFCL"
-					clientManager.CreateClient(context.TODO(), c.Client) // Ignore errors that are caused by duplication
+				identity1 := "LUACWFCL"
+				identity2 := "LUBCWFCL"
+				identity1Sessions, err := mockCreateLoginSessions(m, identity1)
+				identity2Sessions, err := mockCreateLoginSessions(m, identity2)
 
-					lc, _ := MockAuthRequest(fmt.Sprintf("LUACWFCL-%d", i), true)
-					lc.Challenge = fmt.Sprintf("fk-login-challenge-LUACWFCL-%d", i)
-					lc.SessionID = ""
-					require.NoError(t, m.CreateLoginRequest(context.TODO(), lc))
+				require.NoError(t, err)
 
-					require.NoError(t, m.CreateConsentRequest(context.TODO(), c))
-					_, err = m.HandleConsentRequest(context.TODO(), c.Challenge, h)
-					require.NoError(t, err)
+				identities := map[string][]LoginSession{
```

Comment:
I think everything should be created here first, and then  after everything is created, we test it. So as discussed in https://github.com/ory/hydra/pull/1673#discussion_r363869237 - we would set up (including storing it in the store) everything here:

// subject 1 + session 1 + client id 1
// subject 1 + session 2 + client id 2
// subject 2 + session 3 + client id 3
// subject 2 + session 4 + client id 4

```
func createLoginSession(subject, sessionID string) {
  // l := LoginSession{ ...
  // m.CreateLoginSession( ...
}

// ...
// t.Run( ...
createLoginSession(1, 1)
createLoginSession(1, 2)
createLoginSession(2, 3)
createLoginSession(2, 4)

createConsentRequest = func() {
  c = // ...
  lc = MockAuthRequest // ...
  // CreateLoginRequest(lc)
  // ...
  m.CreateConsentRequest(c)
}
```

And then, in a separate loop, would test it like you did here: https://github.com/ory/hydra/pull/1687/files#diff-09be186f7d407a9520f1fdb9316b888aR624